localgovdrupal/localgov_events
# [#161] Delete or archive expired events

## What does this change?

Allows expired events to be deleted or unpublished using a cron job to process the selected action in batches



## How to test

Create some events with some in the past and some in the future and some recurring events spanning both. When the cron job is run, you should see that the correct events are either unpublished or deleted - bearing in mind the number of events to be processed in each cron batch and that the action should only process items which are the specified number of days after the event has expired. You should also see a log of all processed events.




## How can we measure success?

All events can now be unpublished or deleted.



## Have we considered potential risks?

It introduces a method to delete content so there is a risk of unintentional data loss if a user does not configure it correctly.



## Images

Configuration page below:

<img width="1056" height="942" alt="image" src="https://github.com/user-attachments/assets/0dbcef8b-38e5-48cc-a69c-5589a3943129" />



## Accessibility



-   [x] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast) - Not applicable
-   [x] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)  - Not applicable
